File: Assets/_Complete-Game/Scripts/Enemy/EnemyBornList.cs
using UnityEngine;
using UnityEngine.AddressableAssets;
using UnityEngine.ResourceManagement;

public class EnemyBornList : MonoBehaviour
{
    public Transform[] spawnPoints;         // An array of the spawn points enemy can spawn from.

    public static IAsyncOperation Instantiate(AssetReference assetReference, Vector3 position, Quaternion rotation, Transform parent = null)
    {
        return assetReference.Instantiate<GameObject>(position, rotation, parent);
    }
}

File: LuaLib/Assembly-CSharp/EnemyBornList.lua
---@class EnemyBornList : UnityEngine.MonoBehaviour
---@field public spawnPoints UnityEngine.Transform[]
local m = {}

---@overload fun(assetReference:UnityEngine.AddressableAssets.AssetReference, position:UnityEngine.Vector3, rotation:UnityEngine.Quaternion): @static
---@static
---@param assetReference UnityEngine.AddressableAssets.AssetReference
---@param position UnityEngine.Vector3
---@param rotation UnityEngine.Quaternion
---@param parent UnityEngine.Transform
---@return UnityEngine.ResourceManagement.IAsyncOperation
function m.Instantiate(assetReference, position, rotation, parent) end

EnemyBornList = m
return m
